test/e2e-go/features/incentives/mining_test.go

```diff
@@ -0,0 +1,319 @@
+// Copyright (C) 2019-2024 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package suspension
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated/model"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/libgoal"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/framework/fixtures"
+	"github.com/algorand/go-algorand/test/partitiontest"
+)
+
+// first bonus payout, set in config/consensus.go
+const bonus1 = 10_000_000
+
+// TestBasicPayouts shows proposers getting paid
+func TestBasicPayouts(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	defer fixtures.ShutdownSynchronizedTest(t)
+
+	t.Parallel()
+	a := require.New(fixtures.SynchronizedTest(t))
+
+	var fixture fixtures.RestClientFixture
+	// Make the seed lookback shorter, otherwise we need to wait 320 rounds to become IncentiveEligible.
+	fixture.FasterConsensus(protocol.ConsensusFuture)
+	fixture.Setup(t, filepath.Join("nettemplates", "Mining.json"))
+	defer fixture.Shutdown()
+
+	// Overview of this test:
```

Comment:
this is quite complete and complex. I wonder if there is some other e2e test for re-keyreg with with no extra fee (so that it does not become eligible), and checking that when it proposes it does not get any payouts ?

Reply:
Would you be ok with a simpler test that shows that such a re-keyreg does not change `account.IncentiveEligible` to `true`? `TestIncentiveEligible` in `eval_simple_test.go` does that.